pulp_container/app/serializers.py

```diff
@@ -774,11 +773,10 @@ class OCIBuildImageSerializer(ValidateFieldsMixin, serializers.Serializer):
     tag = serializers.CharField(
         required=False, default="latest", help_text="A tag name for the new image being built."
     )
-    artifacts = serializers.JSONField(
+    build_context = RepositoryVersionRelatedField(
         required=False,
-        help_text="A JSON string where each key is an artifact href and the value is it's "
-        "relative path (name) inside the /pulp_working_directory of the build container "
-        "executing the Containerfile.",
+        help_text=_("RepositoryVersion to be used as the build context for container images."),
+        allow_null=True,
```

Comment:
We want this to be a FileRepositoryVersion, right? Can we do something about the queryset here?

Reply:
Hum... I tried to find a `FileRepositoryVersion` model, but I couldn't.
I also tried to pass a `FileRepository` qs, but it didn't work:
```
    build_context = RepositoryVersionRelatedField(
        required=False,
        help_text=_("RepositoryVersion to be used as the build context for container images."),
        allow_null=True,
        queryset = FileRepository.objects.all()
    )


django.core.exceptions.FieldError: Cannot resolve keyword 'number' into field. Choices are: alternatecontentsourcepath, autopublish, content, core_pulp_exporter, description, distributions, group_roles, manifest, name, next_version, pulp_created, pulp_domain, pulp_domain_id, pulp_id, pulp_labels, pulp_last_updated, pulp_type, pulpimporterrepository, remote, remote_id, repository_ptr, repository_ptr_id, repositorycontent, retain_repo_versions, uploads, user_hidden, user_roles, versions
```

I also tried to "adapt" the qs like (to do a reverse lookup of the repository version from repository):
```
        queryset = FileRepository.objects.all().repositoryversion_set,
```
but django complained:
```
AttributeError: 'QuerySet' object has no attribute 'repositoryversion_set'
```

Am I doing something wrong?

Reply:
Maybe it's not possible. But maybe again `RepositoryVersion.objects.filter(repository__pulp_type="file")`?